cmd/skaffold/app/cmd/init.go

```diff
@@ -0,0 +1,312 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/AlecAivazis/survey.v1"
+	yaml "gopkg.in/yaml.v2"
+
+	cmdutil "github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/util"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha2"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+
+	dockerParse "github.com/GoogleContainerTools/kaniko/pkg/dockerfile"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// NoDockerfile allows users to specify they don't want to build
+// an image we parse out from a kubernetes manifest
+const NoDockerfile = "None (image not built from these sources)"
+
+var outfile string
+var skipBuild bool
+var cliArtifacts []string
+
+func NewCmdInit(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Automatically generate skaffold configuration for deploying an application",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return doInit(out)
+		},
+	}
+	AddInitFlags(cmd)
+	return cmd
+}
+
+func AddInitFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&outfile, "file", "f", "", "File to write generated skaffold config")
+	cmd.Flags().BoolVar(&skipBuild, "skip-build", false, "Skip generating build artifacts in skaffold config")
+	cmd.Flags().StringArrayVarP(&cliArtifacts, "artifact", "a", nil, "'='-delimited dockerfile/image pair to generate build artifact\n(example: --artifact=/web/Dockerfile.web=gcr.io/web-project/image)")
+}
+
+func doInit(out io.Writer) error {
+	rootDir := "."
+	var potentialConfigs, k8sConfigs, dockerfiles, images []string
+	err := filepath.Walk(rootDir, func(path string, f os.FileInfo, e error) error {
+		if f.IsDir() {
+			return nil
+		}
+		if util.IsSupportedKubernetesFormat(path) {
```

Comment:
This will return JSON files as well

Reply:
That should be ok. Skaffold will first try and parse it as a Dockerfile (this will fail), then it will try and parse it as a skaffold config (this will also fail), then finally as a kubernetes manifest, which should succeed. I just realized that my code doesn't try and read the k8s manifests as JSON though, let me fix that